tests/test-sources/plugins/pluginmanagers/lz-n.nix

```
@@ -0,0 +1,161 @@
+  */
+  optionalPlugins = map (x: (if (x ? plugin) then x else { plugin = x; }) // { optional = true; });
+in
+{
+  # Empty configuration
+  empty = {
+    plugins.lz-n.enable = true;
+  };
+
+  # single-plugin
+  single-plugin = {
+    module =
+      {
+        config,
+        options,
+        pkgs,
+        lib,
+        helpers,
+        ...
+      }:
+      {
+        extraPlugins = optionalPlugins [ pkgs.vimPlugins.neo-tree-nvim ];
+        plugins.lz-n = {
+          enable = true;
+          settings = {
+            load = lib.mkDefault "vim.cmd.packadd";
+          };
+          plugins = [
+            # enabled, on keys as rawLua
+            {
+              __unkeyed = "neo-tree.nvim";
+              enabled = ''
+                function()
+                return true
+                end
+              '';
+              keys = [
+                {
+                  __unkeyed-1 = "<leader>ft";
+                  __unkeyed-2 = "<CMD>Neotree toggle<CR>";
+                  desc = "NeoTree toggle";
+                }
+              ];
+              after = # lua
+                ''
+                  function()
+                    require("neo-tree").setup()
+                  end
+                '';
+            }
+          ];
+        };
+      };
+  };
+
+  multiple-plugin = {
+    extraPlugins =
+      with pkgs.vimPlugins;
+      [ onedarker-nvim ]
+      ++ (optionalPlugins [
+        neo-tree-nvim
+        dial-nvim
+        vimtex
+        telescope-nvim
+        nvim-biscuits
+        crates-nvim
+      ]);
+    plugins.treesitter.enable = true;
+    plugins.lz-n = {
+      enable = true;
+      plugins = [
+        # enabled, on keys
+        {
+          __unkeyed = "neo-tree.nvim";
+          enabled = ''
+            function()
+            return true
+            end
+          '';
+          keys = [
+            {
+              __unkeyed-1 = "<leader>ft";
+              __unkeyed-2 = "<CMD>Neotree toggle<CR>";
+              desc = "NeoTree toggle";
+            }
+          ];
+          after = # lua
+            ''
+              function()
+                require("neo-tree").setup()
+              end
+            '';
+        }
+        # on keys as list of str and rawLua
+        {
+          __unkeyed = "dial.nvim";
+          keys = [
+            "<C-a>"
+            (helpers.mkRaw ''{ "<C-x>"; mode = "n" }'')
+          ];
+        }
+        # beforeAll, before, on filetype
+        {
+          __unkeyed = "vimtex";
+          ft = [ "plaintex" ];
+          beforeAll = # lua
+            ''
+              function()
+                vim.g.vimtex_compiler_method = "latexrun"
+              end
+            '';
+          before = # lua
+            ''
+              function()
+                vim.g.vimtex_compiler_method = "latexmk"
+              end
+            '';
+        }
+        # On event
+        {
+          __unkeyed = "nvim-biscuits";
```

```suggestion
          __unkeyed-1 = "nvim-biscuits";
```

Same as option declaration and settings-example